tests/unit-tests/src/test/java/org/apache/activemq/artemis/tests/unit/core/remoting/impl/netty/NettyAcceptorTest.java

```diff
@@ -123,4 +124,20 @@ public void testAutoStart() throws Exception {
       assertTrue(server.getRemotingService().getAcceptor("start").isStarted());
       assertFalse(server.getRemotingService().getAcceptor("noStart").isStarted());
    }
+
+   @Test
+   public void testActualPort() throws Exception {
+      String acceptor1Name = RandomUtil.randomString();
+      String acceptor2Name = RandomUtil.randomString();
+      String acceptor3Name = RandomUtil.randomString();
+      ActiveMQServer server = createServer(false, createDefaultInVMConfig());
+      server.getConfiguration().addAcceptorConfiguration(acceptor1Name, "tcp://127.0.0.1:0");
+      server.getConfiguration().addAcceptorConfiguration(acceptor2Name, "tcp://127.0.0.1:0");
+      server.getConfiguration().addAcceptorConfiguration(acceptor3Name, "tcp://127.0.0.1:61616");
+      server.start();
```

Comment:
Are you sure the activation here is not asynchronous? meaning you should use a Wait.assertTrue... or perhaps a Wait.assertTrue(server::isActive) before you check the next assertions?


just trying to avoid an intermittent test failure here due to the async start.

Reply:
Yep I think it will be (thats what prompted my comment about the volatile) so that is a good point.

Reply:
Good point. Change made.